Improve the readme content (#24)
Esri/arcgis-maps-sdk-unreal-engine-samples: sample_project/Content/SampleViewer/Samples/Routing/readme.md

# Find a route

Show a route between two points on a map using Esri's routing service REST API.

![Routing](Routing.jpg)

## How to use the sample

This sample uses Esri's [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm) to query the closest route along the road network between two points. This service uses routing operations associated with your API Key. You can learn more about [API keys](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/) and [Accounts](https://developers.arcgis.com/documentation/mapping-apis-and-services/deployment/accounts/) in the _Mapping APIs and location services_ guide.

1. Open the **Routing** level
2. Click on the **ArcGISMapActor** in the Outliner panel.
3. Set your API key under **Authentication** section in the Details panel.
4. Click play and left-click on two locations on the map while holding shift. The route between the two points will be shown. 

Note: This sample is only set up to work with mouse and keyboard.

## How it works

> 1. The `HTTPClient` is required for using the [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm). 

Reviewer: ```suggestion
1. The `HTTPClient` is used to query the [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm). 
```

> 2. Use the [direct request](https://developers.arcgis.com/rest/network/api-reference/route-synchronous-service.htm) to request and to obtain the routing result.

Reviewer 2: ```suggestion
2. The sample uses the [direct request](https://developers.arcgis.com/rest/network/api-reference/route-synchronous-service.htm) to request and to obtain the routing result.
```

Author: Don't need to comment "Th sample uses" because this is the sample and the section is about how the sample has been made and not a sample description.

Reviewer: "Use" is the imperative mood. Is that what you meant?

I would say

```suggestion
2. The sample makes a [direct request](https://developers.arcgis.com/rest/network/api-reference/route-synchronous-service.htm) to obtain the routing result.
```

Author: Works better

> 3. Use [Traces](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) (Raycasts) to determine the elevation at each breadcrumb's position to account for elevation. (Enable the mesh collider from the **ArcGIS Map Component** to use the Raycast.)

Author: ```suggestion
3. The [Raycast API](https://docs.unrealengine.com/5.0/en-US/API/Runtime/Engine/Engine/UWorld/LineTraceSingleByChannel/) is used to determine the elevation at each breadcrumb's position to account for elevation. (Mesh colliders need to be enabled in the **ArcGIS Map Component** to use the Raycast API)
```

> 3. Use [Traces](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) (Raycasts) to determine the elevation at each breadcrumb's position to account for elevation. (Enable the mesh collider from the **ArcGIS Map Component** to use the Raycast.)

Author: This is what I understood from the source code...

> 3. Use [Traces](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) (Raycasts) to determine the elevation at each breadcrumb's position to account for elevation. (Enable the mesh collider from the **ArcGIS Map Component** to use the Raycast.)

Reviewer 2: ```suggestion
3. Use [Raycast](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) to determine the elevation at each breadcrumb's position to account for elevation. (Enable the mesh collider from the **ArcGIS Map Component** to use the Raycast.)
```

Reviewer 2: better to use the term "raycast" or "line tracing" rather than "traces"

Author: If you see the linked UE doc, they are using `traces` and not `raycast` as the main wording although `raycast` is there since it's included. Also, in the source code, the word `trace` is used instead of `raycast`.

> 3. Use [Traces](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) (Raycasts) to determine the elevation at each breadcrumb's position to account for elevation. (Enable the mesh collider from the **ArcGIS Map Component** to use the Raycast.)

Reviewer: ```suggestion
3. The [Raycast](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) API is used to determine the elevation at each breadcrumb's position to account for elevation. (Mesh colliders need to be enabled in the **ArcGIS Map Component** to use the Raycast API)
```

Author: Raycast API? I don't find the page about it. The one I'm adding is https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/. 
The intention here is to explain the UE feature used in this sample so it's better to follow UE wording for UE things and use our wording for our things.

Reviewer: You can say

```suggestion
3. The [Raycast API](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/) is used to determine the elevation at each breadcrumb's position to account for elevation. (Mesh colliders need to be enabled in the **ArcGIS Map Component** to use the Raycast API)
```

and the link will take the user to it.

"Line Trace" is Unreal's name for "Raycast". I think Game Developers will understand Raycast besides what Apple-like naming Unreal uses.

Author: Apple-like 🍎 😆

> 4. Use the [Spline meshes](https://docs.unrealengine.com/5.0/en-US/BlueprintAPI/SplineMesh/) to visualize the route segments between pairs of breadcrumbs.

Reviewer: same

```suggestion
4. [Spline meshes](https://docs.unrealengine.com/5.0/en-US/BlueprintAPI/SplineMesh/) are used to visualize the route segments between pairs of breadcrumbs.
```

> 

Reviewer 2: a bit nitpicking: there're multiple way to do routing, so rather than say "do this and do that", maybe it's better to say, "this sample does this and that"

Author: That could go just below the How it works but I don't think it's necessary. Users who can try on these advanced samples should be able to figure out other ways if there are any.

Author: I gave a bit more thought to this one. Is it possible that we have other samples that users can obtain the same result in a different way of setting up the scene/level? If so, maybe it's something we can mention on the sample code top page...

Reviewer 2: yes, there're different ways to obtain the same result, but I think we don't need to overexplain...
